### ISSUE TITLE
USE flag in database dump

### ISSUE THREAD

**Issue by pimjansen** [tiredofit/docker-db-backup#249]
### Summary

While importing my database (mysql) it still holds a "USE dbname" flag

### Steps to reproduce

```
- name: BACKUP_LOCATION
  value: blobxfer
- name: BLOBXFER_STORAGE_ACCOUNT
  value: myaccount
- name: BLOBXFER_STORAGE_ACCOUNT_KEY
  value: "****"
- name: BLOBXFER_REMOTE_PATH
  value: /
- name: DB_DUMP_BEGIN
  value: '0300'
- name: TIMEZONE
  value: Europe/Amsterdam
- name: CONTAINER_ENABLE_MONITORING
  value: 'false'
- name: DEBUG_MODE
  value: 'false'
- name: DB_TYPE
  value: mysql
- name: DB_HOST
  value: myhost
- name: DB_NAME
  value: myname
- name: DB_USER
  value: backup
- name: DB_PASS
  value: "***"
- name: DB_CLEANUP_TIME
  value: '604800'
- name: ENABLE_CHECKSUM
  value: 'false'
- name: CHECKSUM
  value: SHA1
- name: COMPRESSION
  value: GZ
- name: SPLIT_DB
  value: 'true'
- name: EXTRA_OPTS
  value: '--complete-insert --no-create-db'
```

### What is the expected *correct* behavior?

While adding the "no-create-db" it is no longer creating the database. However it should also not include an "USE" statement of that database

### Relevant logs and/or screenshots

n/a

### Environment
See reproduce steps

- Image version / tag:  3.9.11
- Host OS: docker

**Comment by ToshY** [comment]
Currently the `SPLIT_DB` variable only refers to databases being dumped to separate archives.

Completely removing `--databases` seems like it would (negatively) impact a lot of users, as `${db_names}` currently takes into consideration `DB_NAME` (which can be multiple databases comma separated) and `DB_NAME_EXCLUDE` (to exclude certain databases). Doesn't seem like a viable option.

@tiredofit Do you have any suggestions?

**Comment by pimjansen** [comment]
> Currently the `SPLIT_DB` variable only refers to databases being dumped to separate archives.
> 
> Completely removing `--databases` seems like it would (negatively) impact a lot of users, as `${db_names}` currently takes into consideration `DB_NAME` (which can be multiple databases comma separated) and `DB_NAME_EXCLUDE` (to exclude certain databases). Doesn't seem like a viable option.
> 
> @tiredofit Do you have any suggestions?

This is not right @ToshY. The --databases flag is there to specify a list of databases for the dump. Which you do not split it, it is not really needed to use it since you only dump a single db in the transaction command

**Comment by tiredofit** [comment]
Since it's in a for loop I think we can get away with dropping that --database argument as we're only backing up one at a time with SPLIT_DB=TRUE..

As a test, pull `docker.io/tiredofit/db-backup:develop-omituse`. I've simply removed the `--database` argument in the backup string and lets see if it gives you what you need. Throughout the day when I find some time I will run this against a series of databases I have on hand to see if any anomalies appear.

**Comment by pimjansen** [comment]
> Since it's in a for loop I think we can get away with dropping that --database argument as we're only backing up one at a time with SPLIT_DB=TRUE..
> 
> 
> 
> As a test, pull `docker.io/tiredofit/db-backup:develop-omituse`. I've simply removed the `--database` argument in the backup string and lets see if it gives you what you need. Throughout the day when I find some time I will run this against a series of databases I have on hand to see if any anomalies appear.

Thanks! Will run it tomorrow morning CET. Already build a local version as well but did not find time to run it. 

So nice that you published a version for this.

I assume it works and understand the dump though since we told it which database it was.

In general it does not make sense to include a create and use schema for a backup since you restrict the usage of the dump. Adding a target to the import command is easy but removing the create and use from a gzipped stream of a few Gb is kinda rough

**Comment by pimjansen** [comment]
I think you broke it @tiredofit 

```
[tiredofit/db-backup 09:38:34 /] $ backup-now
** Performing Manual Backup
/assets/functions/10-db-backup: line 920: syntax error: unexpected end of file
/etc/services.available/10-db-backup/run: line 52: check_availability: command not found
/etc/services.available/10-db-backup/run: line 53: backup_mysql: command not found
2023-09-14.09:38:35 [INFO] ** [db-backup] Backup routines finish time: 2023-09-14 09:38:35 CEST with overall exit code 0
2023-09-14.09:38:35 [NOTICE] ** [db-backup] Backup routines time taken: Hours: 0 Minutes: 00 Seconds: 00
/etc/services.available/10-db-backup/run: line 79: cleanup_old_data: command not found
```

**Comment by pimjansen** [comment]
I open PR for this

**Comment by tiredofit** [comment]
Tagged as `3.10.2` and credited to `pimjansen@github`
